packages/core/src/rules/common/response-contains-header.ts

```diff
@@ -16,9 +16,9 @@ export const ResponseContainsHeader: Oas3Rule | Oas2Rule = (options) => {
             names[getMatchingStatusCodeRange(key).toLowerCase()] ||
             [];
           for (const expectedHeader of expectedHeaders) {
-            if (!response.headers?.[expectedHeader]) {
+            if (!response.headers?.[expectedHeader.toLowerCase()]) {
```

Comment:
I believe, this won't cover the case when `response.headers` keys are not in lower case itself. E.g.: 
```
response.headers == { foo: 1, BAR: 2 }
```

It looks like lowercasing the keys should help better: 
```ts
Object.keys(response.headers).some(key => key.toLowerCase() === expectedHeader.toLowerCase())
```


Also, don't you mind adding a test?